source/agora/consensus/data/Transaction.d

```diff
@@ -159,6 +136,29 @@ public struct Transaction
     {
         return hashFull(this).opCmp(hashFull(other));
     }
+
+    pure nothrow @nogc:
+
+    /// A `Freeze` transaction is one that has one or more `Freeze` outputs
+    /// If there is more than one output then it is allowed to have a single
+    ///  `Payment` output for a refund of any amount
+    public bool isFreeze () const
+    {
+        return this.outputs.any!(o => o.type == OutputType.Freeze);
+    }
+
+    /// A `Coinbase` transaction is one that has one or more `Coinbase` outputs
+    /// However if all outputs are not `Coinbase` then it will fail validation
+    public bool isCoinbase () const
+    {
+        return this.outputs.any!(o => o.type == OutputType.Coinbase);
+    }
+
+    /// A `Payment` transaction is one that has outputs of type `Payment`
+    public bool isPayment () const
+    {
+        return this.outputs.all!(o => o.type == OutputType.Payment);
+    }
```

Comment:
@MichaelKim20  I added some more documentation here about the Transaction and output type relationship.